<!-- Launch file for the rrt-detector and the assigner -->


<launch>
<arg name="eta" value="1.0"/>
<arg name="Geta" value="1.0"/>
<arg name="info_radius" value="1.0"/>
<arg name="info_multiplier" value="3.0"/>
<arg name="hysteresis_radius" value="3.0"/>
<arg name="hysteresis_gain" value="2.0"/>
<arg name="costmap_clearing_threshold" value="50"/>
<arg name="delay_after_assignement" value="0.5"/>
<arg name="frequency" value="50"/>


<arg name="namespace" value="tb3_1"/>

<param name="namespace_init_count" value="1"/>


  <node pkg="rrt_exploration" type="global_rrt_detector" name="global_detector" output="screen">
  <param name="eta" value="$(arg Geta)"/>
  <param name="map_topic" value="$(arg namespace)/map"/>
  <!-- <param name="map_topic" value="/projected_map"/> -->
  </node>
  
  <node pkg="rrt_exploration" type="local_rrt_detector" name="local_detector" output="screen">
  <param name="eta" value="$(arg eta)"/>
  <param name="map_topic" value="$(arg namespace)/map"/>
  <!-- <param name="map_topic" value="/projected_map"/> -->
  <param name="robot_frame" value="$(arg namespace)/base_link"/>
  </node>
 
  
  
  <!-- <node pkg="frontier_opencv_detector" type="local_rrt_detector" name="local_detector" output="screen">
  <param name="eta" value="$(arg eta)"/>
  <param name="map_topic" value="$(arg namespace)/map"/>
  <param name="robot_frame" value="$(arg namespace)/base_link"/>
  </node> -->
 
  <node pkg="rrt_exploration" type="filter.py" name="filter" output="screen">
  <param name="map_topic" value="$(arg namespace)/map"/>
  <param name="info_radius" value="$(arg info_radius)"/> 
  <param name="costmap_clearing_threshold" value="$(arg costmap_clearing_threshold)"/> 
  <param name="goals_topic" value="/detected_points"/>
  <param name="namespace" value="tb3_"/> 
  <param name="n_robots" value="1"/>
  <param name="rate" value="$(arg frequency)"/>
  </node>
  
  <node pkg="rrt_exploration" type="assigner.py" name="assigner" output="screen">
  <param name="map_topic" value="$(arg namespace)/map"/>
  <param name="global_frame" value="map"/>
  <param name="info_radius" value="$(arg info_radius)"/> 
  <param name="info_multiplier" value="$(arg info_multiplier)"/> 
  <param name="hysteresis_radius" value="$(arg hysteresis_radius)"/> 
  <param name="hysteresis_gain" value="$(arg hysteresis_gain)"/> 
  <param name="frontiers_topic" value="/filtered_points"/> 
  <param name="n_robots" value="1"/>
  <param name="namespace" value="tb3_"/>
  <param name="delay_after_assignement" value="$(arg delay_after_assignement)"/>
  <param name="rate" value="$(arg frequency)"/>
  </node>
  
</launch>
